addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+    private ValloxMVWebSocketListener socket;
+
+    private final Logger logger = LoggerFactory.getLogger(ValloxMVWebSocket.class);
+
+    public ValloxMVWebSocket(ValloxMVHandler voHandler, String ip) {
+        this.voHandler = voHandler;
+        this.ip = ip;
+        client = new WebSocketClient();
+    }
+
+    public void request(ChannelUID channelUID, String updateState) {
+        try {
+            socket = new ValloxMVWebSocketListener(channelUID, updateState);
+            client.start();
+
+            URI destUri = new URI("ws://" + ip + ":80");
+
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            logger.debug("Connecting to: {}", destUri);
+            client.connect(socket, destUri, request);
+            socket.awaitClose(2, TimeUnit.SECONDS);
+        } catch (IOException e) {
+            connectionError(e);
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage());
+        }
+    }
+
+    public void close() {
+        try {
+            client.stop();
+        } catch (IOException e) {
+            connectionError(e);
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage());
+        }
+        client.destroy();
+        client = null;
+    }
+
+    public void connectionError(IOException e) {
+        logger.error("Error connecting vallox unit.", e);
+        voHandler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR);
+    }
+
+    @WebSocket
+    public class ValloxMVWebSocketListener {
+        private final CountDownLatch closeLatch = new CountDownLatch(1);
+
+        private Logger logger = LoggerFactory.getLogger(ValloxMVWebSocket.class);
+        private String updateState;
+        private ChannelUID channelUID;
+
+        public ValloxMVWebSocketListener(ChannelUID channelUID, String updateState) {
+            this.updateState = updateState;
+            this.channelUID = channelUID;
+        }
+
+        @OnWebSocketConnect
+        public void onConnect(Session session) {
+            try {
+                logger.debug("Connect: {}", session.getRemoteAddress().getAddress());
+                ByteBuffer buf = generateRequest();
+                session.getRemote().sendBytes(buf);
+            } catch (IOException e) {
+                connectionError(e);
+            } catch (Exception e) {
+                logger.error("Error: {}", e.getMessage());
+            }
+        }
+
+        /**
+         * Method to generate ByteBuffer request to be sent to vallox online websocket
+         * to request or set data
+         *
+         * @param mode 246 for data request, 249 for setting data
+         * @param hmParameters HashMap for setting data with register as key and value as value
+         * @return ByteBuffer to be sent to websocket
+         */
+        public ByteBuffer generateCustomRequest(Integer mode, HashMap<Integer, Integer> hmParameters) {
```

Comment:
For method parameters, return types, fields etc, always use the interface of the Java collection types instead of specific implementation, in this case just use `Map<...>`